app/types/ResourceEntry.ts

```diff
@@ -0,0 +1,195 @@
+/**
+ * PHLask Resource Entry Types
+ *
+ * These types define the schema for PHLask resources stored in the Supabase database.
+ * Based on the schema from the phlask-map project.
+ */
+
+/**
+ * A data source defining where the resource data entry came from.
+ */
+export interface DataSource {
+  /** The type of data source */
+  type: 'MANUAL' | 'WEB_SCRAPE';
```

Comment:
for union types: can we declare them separately? I can see cases where that we might want to type certain data separately if we submit it within the app. That will also be cleaner for arrays of union types